cmd/buildkitd/main.go

```diff
@@ -310,9 +310,13 @@ func setDefaultConfig(cfg *config.Config) {
 
 	if cfg.Workers.OCI.Platforms == nil {
 		cfg.Workers.OCI.Platforms = binfmt_misc.SupportedPlatforms()
+	} else {
+		binfmt_misc.ValidatePlatforms(cfg.Workers.OCI.Platforms)
 	}
 	if cfg.Workers.Containerd.Platforms == nil {
 		cfg.Workers.Containerd.Platforms = binfmt_misc.SupportedPlatforms()
+	} else {
```

Comment:
just run it once per each worker instead of looking for else/flags-set case. if detection was automatic validation is just a no-op.